package/MDAnalysis/analysis/rms.py

```diff
@@ -488,6 +495,15 @@ def __init__(self, atomgroup, reference=None, select='all',
                     "N_ref={3}, N_traj={4}".format(
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
+        # check weights
+        if not (iterable(self.weights) or self.weights in ("mass", None)):
+            raise TypeError("weights must be {'mass', None} or an iterable of the "
+                            "same size as the atomgroup.")
+        elif iterable(self.weights) and \
+             len(self.weights) != self.mobile_atoms.n_atoms:
+            raise TypeError("weights (length {0}) must be of same length as "
+                            "the mobile selection ({1})".format(
+                                len(self.weights), self.mobile_atoms.n_atoms))
```

Comment:
we have weights checks several times in the analysis modules now. Do you mind factoring this into a function and place it in lib so we can use these checks in more places. It's quite useful.